status/flowcell.py

```diff
@@ -592,3 +588,22 @@ def add_prefix(input_int: int, unit: str):
 
     output_str = " ".join([str(output_int), unit])
     return output_str
+
+
+def walk_str2int(iterable):
+    """This function recursively traverses a JSON-like tree of mutable iterables (dicts, lists)
+    and reassigns all list elements or dict values which are strings that can be intepreted as integers
+    to the int type"""
+
+    if isinstance(iterable, (dict, list)):
+        for key, val in (
+            iterable.items() if isinstance(iterable, dict) else enumerate(iterable)
+        ):
+            if isinstance(val, str) and re.match("^\d+$", val):
+                iterable[key] = int(val)
+            elif isinstance(val, (dict, list)):
+                walk_str2int(val)
+            else:
+                continue
```

Comment:
This is implicit, you don't need to write it out, but couldn't hurt I guess.

Reply:
Yes, I thought the explicit logical outcomes would ease your mind 😆